tests/components/mqtt/conftest.py

```diff
@@ -10,3 +10,15 @@ def mqtt_mock(loop, hass):
     client = loop.run_until_complete(async_mock_mqtt_component(hass))
     client.reset_mock()
     return client
+
+
+@pytest.fixture(autouse=True)
+def fail_on_log_exception(request, monkeypatch):
+    """Fixture to fail if MQTT callbacks throw."""
+    if "no_fail_on_log_exception" in request.keywords:
+        return
+
+    def log_exception(format_err, *args):
+        pytest.fail()
+
+    monkeypatch.setattr("homeassistant.util.logging.log_exception", log_exception)
```

Comment:
We already have this built into all tests https://github.com/home-assistant/core/blob/dev/tests/conftest.py#L84-L100